### PR TITLE
Remove unneeded BFEs around flag generation code.

### PR DESCRIPTION
**Pull request by phire**
Depends on #365. Only last few commits are relevant.

### CODE REVIEW

**Comment by phire** [comment]
Looks like this has broken Nvidia drivers. 

Things like glxgears and FTL are just showing a black window.

Will track down later.

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 1]
```diff
@@ -1772,11 +1788,14 @@ void OpDispatchBuilder::ASHRImmediateOp(OpcodeArgs) {
     Shift &= 0x1F;
 
   OrderedNode *Src = _Constant(Size, Shift);
-  auto ALUOp = _Ashr(Dest, Src);
+  OrderedNode *Result = _Ashr(Dest, Src);
```
Comment (phire) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1791]: Also need sign extend

**Comment by phire** [comment]
Rebased and tracked down the nvidia driver issue (INC overflow had incorrect zero flag)

Fixed and added unit tests

**Review by Sonicadvance1 — APPROVED** [review]
Changes SHLD behaviour slightly for zero shifts, but it is closer to correct so still a win.

**Pull request merged**